sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusEndToEndTests.cs

```diff
@@ -196,13 +196,28 @@ public async Task TestBatch_JsonPoco()
         [Test]
         public async Task TestSingle_JObject()
         {
-            var (jobHost, _) = BuildHost<ServiceBusMultipleMessagesTestJob_BindToJObject>();
+            var (jobHost, host) = BuildHost<ServiceBusMultipleMessagesTestJob_BindToJObject>();
             using (jobHost)
             {
                 await WriteQueueMessage(JsonConvert.SerializeObject(new {Date = DateTimeOffset.Now}));
                 bool result = _eventWait.WaitOne(SBTimeoutMills);
                 Assert.True(result);
                 await jobHost.StopAsync();
+                AssertNoLoggedErrors(host);
+            }
+        }
+
+        [Test]
+        public async Task TestBatch_NoMessages()
+        {
+            var (jobHost, host) = BuildHost<ServiceBusMultipleMessagesTestJob_NoMessagesExpected>();
+            using (jobHost)
+            {
+                // wait for 20 seconds to let the listener run for a while
```

Comment:
![https://media3.giphy.com/media/7p3e2WCM0VEnm/200.gif](https://media3.giphy.com/media/7p3e2WCM0VEnm/200.gif)

Reply:
can we tune some parameters so we don't need to wait 20 sec?

Reply:
What code-path executed the ServiceBusMultipleMessagesTestJob_NoMessagesExpected before? Why does it take 20 sec to be hit?

Reply:
This is a new test. I was being conservative with 20 seconds to account for the host startup.